src/renderers/common/Renderer.js

```diff
@@ -1584,23 +1584,23 @@ class Renderer {
 
 						this.backend.updateViewport( this._currentRenderContext );
 
-						this._currentRenderObjectFunction( object, scene, camera2, geometry, material, group, lightsNode, passId );
+						this._currentRenderObjectFunction( object, scene, camera2, geometry, material, group, lightsNode, clippingContext, passId );
 
 					}
 
 				}
 
 			} else {
 
-				this._currentRenderObjectFunction( object, scene, camera, geometry, material, group, lightsNode, passId );
+				this._currentRenderObjectFunction( object, scene, camera, geometry, material, group, lightsNode, clippingContext, passId );
 
 			}
 
 		}
 
 	}
 
-	renderObject( object, scene, camera, geometry, material, group, lightsNode, passId = null ) {
+	renderObject( object, scene, camera, geometry, material, group, lightsNode, clippingContext = null, passId = null ) {
```

Comment:
`LensflareMesh` directly calls `renderObject()` and can't access `clippingContext` meaning it can't pass it to `renderObject()`. I've fixed the resulting runtime error by defining a default parameter.

If we add `clippingContext` to `onBeforeRender()`, we could ensure lensflares support clipping as well. Alternatively, we could also use the "current clipping context" for nested renderings if it isn't overwritten with a parameter when calling `renderObject()`.